pkg/plugins/local_test.go

```diff
@@ -103,3 +106,42 @@ func testInDir(t *testing.T, casename, dir string, tmps, defexp []types.Capabili
 		assert.Equal(t, defexp, gotDef, casename)
 	}
 }
+
+func TestRemoveLegacyTemps(t *testing.T) {
+	dir := "vela-test-rm-temps"
+
+	cases := []struct {
+		caseName string
+		newTemps []types.Capability
+		rmNum    int
+	}{
+		{
+			caseName: "remove all",
+			newTemps: []types.Capability{},
+			rmNum:    3,
+		},
+		{
+			caseName: "nothing removed",
+			newTemps: []types.Capability{deployment, statefulset, route},
+			rmNum:    0,
+		},
+		{
+			caseName: "remove part of existings",
+			newTemps: []types.Capability{statefulset, route},
+			rmNum:    1,
+		},
+	}
+
+	for _, c := range cases {
+		func(t *testing.T) {
+			err := os.MkdirAll(dir, 0755)
+			assert.NoError(t, err, c.caseName)
+			defer os.RemoveAll(dir)
+			existingTemps := []types.Capability{deployment, statefulset, route}
+			number := SinkTemp2Local(existingTemps, dir)
+			assert.Equal(t, 3, number)
+			rmNum := RemoveLegacyTemps(c.newTemps, dir)
+			assert.Equal(t, c.rmNum, rmNum, c.caseName)
+		}(t)
```

Comment:
writing a function inside the test loop makes  it hard to read, I suggest to separate it out, or don't use defer.

Reply:
got it.